### DISCUSSION TITLE
Continuous integration suggestions

### DISCUSSION THREAD

**Discussion by RedDragonWebDesign** [pdfarranger/pdfarranger#968] [General]
* [ ] Suggest having CI run on every commit. Currently looks like it only runs on some commits. This can prevent bugs where the logic of what files to run CI for is wrong.
* [ ] Suggest removing the codecov tests from the "commit" action, and just run them on the PR action. That way you don't get red X's in your list of commits. Right now it looks like half of commits are red X's, which could lead folks to ignore CI completely because it is generating too much noise/false positives.

Thanks for being one of the only free and good PDF editors I've found :)

**Comment by dreua** [comment]
I think the logic of our CI works fine, it runs on the last commit of each PR and additionally on main. Running it for each and every commit seems like a waste of resources. I don't think it is necessary for this project to have each and every commit checked and green.

Do you have examples for where the current might be wrong?

The codecov runs on the same schedule so if we don't change that, it should essentially do what you suggest. It does, however, have other issues therefore I just created #995 .
